tools/SourceKit/tools/sourcekitd/lib/API/CMakeLists.txt

```diff
@@ -16,5 +16,8 @@ endif()
 
 add_sourcekit_library(sourcekitdAPI
   ${sourcekitdAPI_sources}
-  DEPENDS SourceKitSupport
+  DEPENDS
+      SourceKitSupport SourceKitSwiftLang
+      # Clang dependencies, required by SourceKitSwiftLang
+      clangFormat clangToolingCore clangDriver
```

Comment:
You shouldn't have to do this. CMake understands transitive dependencies between static archives, it should do the right thing.

I suspect this is caused by the way Swift builds out-of-tree with LLVM & Clang. My "stab in the dark" here is that it handles static libraries specified by full path differently than libraries that are specified by target.

I wonder if this could be fixed in a minimally invasive way by having Swift include the CMake Exports lists from Clang and LLVM. Then the LLVM and Clang targets would be exposed more cleanly in Swift's build.